src/main.rs

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use futures::StreamExt;
+use std::io;
+use typedb_client::{
+    concept::{Attribute, Concept, Value},
+    Connection, DatabaseManager, Session,
+    SessionType::Data,
+    TransactionType::Read,
+};
+mod common;
+
+#[derive(Debug)]
+enum HandleError {
+    Io(io::Error),
+    TypeDB(typedb_client::error::Error),
+}
+
+const MENU_DATABASE: &str = "menuDB1";
+fn read_input() -> Result<String, HandleError> {
+    let mut input_string = String::new();
+    io::stdin()
+        .read_line(&mut input_string)
+        .map_err(HandleError::Io)?;
+    Ok(input_string.trim().to_string())
+}
+
+async fn get_sellers(connection: Connection) -> Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(
+        databases
+            .get(MENU_DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?,
+        Data,
+    )
+    .await
+    .map_err(HandleError::TypeDB)?;
+    let transaction = session
+        .transaction(Read)
+        .await
+        .map_err(HandleError::TypeDB)?;
+
+    println!("Enter the name of raw_food x : ");
+    let x: String = read_input()?;
+    let q = format!(
+        "match $rf isa raw_food, has name \"{x}\";
+        $is_i (raw_food: $rf,dish: $d) isa is_ingredient;
+        $sl (seller: $m,product: $d) isa sells;
+        $m has name $n;
+        get $n;"
+    );
+    let mut answer_stream = transaction
+        .query()
+        .match_(q.as_str())
+        .map_err(HandleError::TypeDB)?;
+    while let Some(result) = answer_stream.next().await {
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute {
+                        value: Value::String(value),
+                        ..
+                    }) = concept
+                    {
+                        println!("{}", value);
+                    }
+                }
+            }
+            Err(err) => {
+                return Err(HandleError::TypeDB(err));
+            }
+        }
+    }
```

Comment:
We use this fairly long block in both `get_sellers` and `get_strange_menu`. Let's deduplicate this by moving it to a function that prints out string answers from the concept map.